homeassistant/components/nest/config_flow.py

```diff
@@ -199,6 +232,111 @@ async def async_step_auth(
             )
         return await super().async_step_auth(user_input)
 
+    def _configure_pubsub(self) -> bool:
+        """Return True if the config flow should configure Pub/Sub."""
+        if self._reauth_data is not None and CONF_SUBSCRIBER_ID in self._reauth_data:
+            # Existing entry needs to be reconfigured
+            return True
+        if CONF_SUBSCRIBER_ID in self.hass.data[DOMAIN][DATA_NEST_CONFIG]:
+            # Hard coded configuration.yaml skips pubsub in config flow
+            return False
+        # No existing subscription configured, so create in config flow
+        return True
+
+    async def async_step_pubsub(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Configure and create Pub/Sub subscriber."""
+        # Populate data from the previous config entry during reauth, then
+        # overwrite with the user entered values.
+        cloud_project_id = ""
+        subscriber_id = ""
+        for data in (self._reauth_data, user_input):
```

Comment:
It's a bit of weird code because `user_input` can only ever contain `CONF_CLOUD_PROJECT_ID`. You just want the merged dictionaries if either are defined right?

This might be more readable:

```python
data = {}
if self._reauth_data:
  data.update(self._reauth_data)
if user_input:
  data.update(user_input)
```

Reply:
Yes, that is simpler. I still feel like this this function is more complex than it needs to be, but this is getting better.